lib/rspec/mocks/message_expectation.rb

```diff
@@ -98,7 +99,7 @@ def and_return(first_value, *values)
       #   expect(counter.count).to eq(original_count + 1)
       def and_call_original
         wrap_original(__method__) do |original, *args, &block|
-          original.call(*args, &block)
+          RSpec::Support::WithKeywordsWhenNeeded.call(original, *args, &block)
```

Comment:
This can't be used, it causes problems with arguments that implement `to_hash` and this module is being phased out for removal in RSpec 4.